luarules/gadgets/unit_weapon_smart_select_helper.lua

```diff
@@ -0,0 +1,217 @@
+function gadget:GetInfo()
+	return {
+		name = "Weapon Smart Select",
+		desc = "Integrates with smart_weapon_select.h and animation scripts to switch firing modes.",
+		author = "SethDGamre",
+		date = "2024.12.7",
+		license = "GNU GPL, v2 or later",
+		layer = 2,
+		enabled = true
+	}
+end
+
+--increment this counter for every hour of your life wasted on smart select: 52
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+--[[
+Integration Checklist:
+1. use weapondef.customparams.smart_priority = true for the higher priority smart select weapon.
+2. use weapondef.customparams.smart_backup = true for the fallback weapon when smart_backup doesn't have a target, or cannot shoot the manually selected target.
+3. use weapondef.customparams.smart_trajectory_checker = true for the weapon that should be used for trajectory checks for the priorityWeapon. Ideally this is a static point slightly lower than preferred_weapon
+3. in the unit's .bos animation script, #include "smart_weapon_select.h"  ideally at the beginning of the file.
+4. in the preferred AimWeaponX() function, add the following at the beginning:
+	if (AimingState != AIMING_PRIORITY){
+		return(0);
+	}
+5. in the deferred AimWeaponX() function, add the following at the beginning:
+	if (AimingState != AIMING_BACKUP){
+		return(0);
+	}
+6. If using a dummy weapon, return (0); in its AimWeaponX() function and QueryWeaponX(piecenum) should be set to static piece lower than the turret.
+
+****OPTIONAL*****
+use weapondef.customparams.smart_error_frames to override the default reloadtime derivitive frames error threshold
```

Comment:
The turn speed of turrets may be abnormally slow, which would cause nuisance misfires. The turn speed of the turret is only defined in the AimWeaponX callin in the animation script and as such isn't accessible without rewriting those functions. Easier to just give them a customparam to calibrate the duration if necessary. (so far, not necessary)
It's there so Zecrus or Johannas can add it later without requesting a gadgeteer such as myself to do so later.